_modules/azure.js

```diff
@@ -0,0 +1,58 @@
+// -----------------------------------------------------------------------------------------
+// #region Imports
+// -----------------------------------------------------------------------------------------
+
+const echo  = require("./echo");
+const shell = require("shelljs");
+
+// -----------------------------------------------------------------------------------------
+// #region Functions
+// -----------------------------------------------------------------------------------------
+
+const azure = {
+    /**
+     * Executes `az login` using username/password.
+     * @param {*} username Azure account username
+     * @param {*} secret Azure account password
+     */
+    login(username, secret) {
+        const loginCommand = `az login -u ${username} -p ${secret}`;
+        if (shell.exec(loginCommand).code !== 0) {
+            echo.error(" - Failed to login to Azure");
+            shell.exit(1);
+        }
+    },
+    /**
+     * Executes `az login` using service principal.
+     * @param {*} clientId Client ID for the service principal
+     * @param {*} tenantId Tenant ID for the service principal
+     * @param {*} secret Secret for the service principal
+     */
+    login(clientId, tenantId, secret) {
+        const loginCommand = `az login --service-principal -u ${clientId} -t ${tenantId} -p=${secret}`;
+        if (shell.exec(loginCommand).code !== 0) {
```

Comment:
Suggestion - pull out the result of `shell.exec(loginCommand)` so that the `code` can be checked and passed along to `shell.exit()` so that information is not obscured when returning to the caller/environment (ie Jenkins, your local shell).